lib/created_message_queue_handler.ts

```diff
@@ -415,32 +398,50 @@ export async function index(
       defaultAddresses
     );
 
-    const shouldTriggerARetryOrOutputBindings = await errorOrNotification.fold(
-      error => processRuntimeError(queueService, error, context.bindingData),
-      notification =>
-        processSuccess(
-          notificationStatusModel,
-          notification,
-          newMessageWithContent,
-          senderMetadata
-        )
+    if (isLeft(errorOrNotification)) {
+      // something went wrong trying to get the notification object
+      // if it's a transient error, this will trigger a retry
+      throw errorOrNotification.value;
+    }
+    const notification = errorOrNotification.value;
+
+    // try to update the message status in the database
+    const errorOrUpdatedMessageStatus = await messageStatusUpdater(
+      MessageStatusValueEnum.PROCESSED
     );
+    if (isLeft(errorOrUpdatedMessageStatus)) {
+      // this will trigger a retry as it's considered a transient error
+      throw errorOrUpdatedMessageStatus.value;
+    }
 
-    return shouldTriggerARetryOrOutputBindings.fold(
-      shouldTriggerARetry => {
-        if (shouldTriggerARetry) {
-          throw TransientError("retry");
-        }
-        return stopProcessing;
+    // everything went well, make output bindings
+    const emailNotification: NotificationEvent = {
+      message: {
+        ...newMessageWithContent,
+        kind: "INewMessageWithContent"
       },
-      // success ! lets see if we have some bindings to output
-      outputBindings => outputBindings.toUndefined()
+      notificationId: notification.id,
+      senderMetadata
+    };
+
+    winston.debug(
+      `CreatedMessageQueueHandler|succeeded|message=${
+        emailNotification.notificationId
+      }|message=${emailNotification.message.id}`
     );
+
+    return { emailNotification };
+    //
   } catch (error) {
```

Comment:
we cannot just return eithers and map here as at some point we 
*must* catch unknow exceptions which may occur inside library code